Wikipedia/Code/TalkPageFetcher.swift

```diff
@@ -64,5 +64,79 @@ class TalkPageFetcher: Fetcher {
             }
         }
     }
-
+    
+    func subscribeToTopic(talkPageTitle: String, siteURL: URL, topic: String, shouldSubscribe: Bool, completion: @escaping (Result<Bool, Error>) -> Void) {
+        
+        guard let title = talkPageTitle.denormalizedPageTitle else {
+            completion(.failure(RequestError.invalidParameters))
+            return
+        }
+                
+        var params = ["action": "discussiontoolssubscribe",
+                      "page": title,
+                      "format": "json",
+                      "commentname": topic,
+                      "formatversion": "2"
+        ]
+        
+        if shouldSubscribe {
+            params["subscribe"] = "1"
+        }
+        
+        performTokenizedMediaWikiAPIPOST(to: siteURL, with: params, reattemptLoginOn401Response: true) { result, httpResponse, error in
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+            
+            if let resultError = result?["error"] as? [String: Any],
+               let info = resultError["info"] as? String {
+                completion(.failure(RequestError.api(info)))
+                return
+            }
+            
+            if let resultSuccess = result?["discussiontoolssubscribe"] as? [String: Any],
+                let didSubscribe = resultSuccess["subscribe"] as? Bool {
+                completion(.success(didSubscribe))
+                return
+            }
+            completion(.failure(RequestError.unexpectedResponse))
+        }
+    }
+    
+    func getAllSubscriptions(siteURL: URL, topics: [String], completion: @escaping (Result<[String], Error>) -> Void) {
+        
+        let joinedString = topics.joined(separator: "|")
+        
+        let params = ["action": "discussiontoolsgetsubscriptions",
+                       "format": "json",
+                       "commentname": joinedString,
+                       "formatversion": "2"
+        ]
+        
+        performMediaWikiAPIGET(for: siteURL, with: params, cancellationKey: nil) { result, httpResponse, error in
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+            
+            if let resultError = result?["error"] as? [String: Any],
+               let info = resultError["info"] as? String {
+                completion(.failure(RequestError.api(info)))
+                return
+            }
+            
+            if let resultSuccess = result?["subscriptions"] as? [String: Any] {
+                var subscribedTopics = [String]()
+                for (topicId, _) in resultSuccess {
```

Comment:
I came across a situation with the API where the response was this:

```
{
  "subscriptions": {
    "h-Tsevener-20220720150400": 0,
    "h-TSevener_(WMF)-2022-06-24T18:39:00.000Z": 1
  }
}
```

I had to first make the **unsubscribe** call to `h-Tsevener-20220720150400`, then it started returning with a `0` value in this response. So I think you should check the dictionary value here as well to make sure it's a `1` value, before appending to `subscribedTopics`.